openlibrary/macros/AvailabilityButton.html

```diff
@@ -1,8 +1,8 @@
 $def with(page, availability, user)
 
 $ ocaid = availability and availability.get('identifier')
-$ loan = ocaid and [loan for loan in user['loans'] if ocaid == loan.get('ocaid')]
-$ waiting_loan = ocaid and [request for request in user['waitlists'] if ocaid == request.get('identifier')]
+$ loan = ocaid and user and [loan for loan in user.get('loans') if loan and ocaid == loan.get('ocaid')]
```

Comment:
The boolean logic has become a little complex/hard to grok; this might help make it easier to understand.

```suggestion
$ loan = waiting_loan = None
$if ocaid and user:
  $ loans = [loan for loan in user.get('loans') if loan and ocaid == loan.get('ocaid')]
  $ loan = loans and loans[0]
  $ waiting_loans = [request for request in user.get('waitlists') if request and ocaid == request.get('identifier')]
  $ waiting_loan = waiting_loans and waiting_loans[0]
$ loan = ocaid and user and [loan for loan in user.get('loans') if loan and ocaid == loan.get('ocaid')]
```

Reply:
I'd rather not make this change as if `ocaid` or `user` are not set then it looks like `waiting_loan` wouldn't be defined (as `None` or otherwise) and it is referenced further down in the file:

https://github.com/internetarchive/openlibrary/blob/1ab475b97ed0d27c2165a64a3b59511adcfab8f6/openlibrary/macros/AvailabilityButton.html#L31